internal/querynodev2/segments/result.go

```diff
@@ -479,18 +482,23 @@ func MergeSegcoreRetrieveResults(ctx context.Context, retrieveResults []*segcore
 		ts := validRetrieveResults[sel].Timestamps[cursors[sel]]
 		if _, ok := idTsMap[pk]; !ok {
 			typeutil.AppendPKs(ret.Ids, pk)
-			selected = append(selected, sel)
-			selectedOffsets[sel] = append(selectedOffsets[sel], validRetrieveResults[sel].Result.GetOffset()[cursors[sel]])
-			selectedIndexes[sel] = append(selectedIndexes[sel], cursors[sel])
+			selections = append(selections, selection{
+				batchIndex:  sel,
+				resultIndex: cursors[sel],
+				offset:      validRetrieveResults[sel].Result.GetOffset()[cursors[sel]],
+			})
 			idTsMap[pk] = ts
 			availableCount++
 		} else {
 			// primary keys duplicate
 			skipDupCnt++
 			if ts != 0 && ts > idTsMap[pk] {
-				idTsMap[pk] = ts
-				selectedOffsets[sel][len(selectedOffsets[sel])-1] = validRetrieveResults[sel].Result.GetOffset()[cursors[sel]]
-				selectedIndexes[sel][len(selectedIndexes[sel])-1] = cursors[sel]
+				lastIdx := len(selections) - 1
+				selections[lastIdx] = selection{
```

Comment:
Very strange, the pk of last selection may be different with the newest selection.

Reply:
I think we can just remove this code branch (or comment them).

Reply:
The last selection shall be the same PK due to our result sorting convention that: all result shall be sorted by PK. Otherwise all reduce logic will be broken

Reply:
If everything works as expected, then the else branch makes no sense neither, the same pk with older timestamp can't be even selected.

Reply:
Agree, but we shall make the logic correct here